src/builder/create_interaction_response.rs

```diff
@@ -54,35 +54,80 @@ pub enum CreateInteractionResponse {
     Modal(CreateModal),
 }
 
+#[derive(Serialize, Deserialize)]
+struct InteractionResponseJson {
+    #[serde(rename = "type")]
+    kind: u8,
+    data: serde_json::Value,
+}
+
 impl serde::Serialize for CreateInteractionResponse {
-    fn serialize<S: serde::Serializer>(&self, serializer: S) -> StdResult<S::Ok, S::Error> {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         use serde::ser::Error as _;
 
         #[allow(clippy::match_same_arms)] // hurts readability
-        serde_json::json!({
-            "type": match self {
-                Self::Pong { .. } => 1,
-                Self::Message { .. } => 4,
-                Self::Defer { .. } => 5,
-                Self::Acknowledge { .. } => 6,
-                Self::UpdateMessage { .. } => 7,
-                Self::Autocomplete { .. } => 8,
-                Self::Modal { .. } => 9,
+        InteractionResponseJson {
+            kind: match self {
+                Self::Pong {
+                    ..
+                } => 1,
+                Self::Message {
+                    ..
+                } => 4,
+                Self::Defer {
+                    ..
+                } => 5,
+                Self::Acknowledge {
+                    ..
+                } => 6,
+                Self::UpdateMessage {
+                    ..
+                } => 7,
+                Self::Autocomplete {
+                    ..
+                } => 8,
+                Self::Modal {
+                    ..
+                } => 9,
```

Comment:
```suggestion
                Self::Pong => 1,
                Self::Message(_) => 4,
                Self::Defer(_) => 5,
                Self::Acknowledge => 6,
                Self::UpdateMessage(_) => 7,
                Self::Autocomplete(_) => 8,
                Self::Modal(_) => 9,
```

Weird match patterns here that got exposed by rustfmt once it wasn't wrapped in a macro call.